File: greendaoMultiTable/src/main/java/com/greendao/multitable/DBManager.java
package com.greendao.multitable;


import com.greendao.multitable.bean.test.DaoMaster;
import com.greendao.multitable.bean.test.DaoSession;

/**
 * by y on 05/07/2017.
 */

public class DBManager {

    private static final String SQL_NAME = "sample";

    private DBManager() {
    }

    public static DaoSession getDaoSession() {
        return SessionHolder.daoSession;
    }

    private static class SessionHolder {
        private static final DaoSession daoSession = new DaoMaster(new DaoMaster.DevOpenHelper(App.getContext(), SQL_NAME, null).getWritableDatabase()).newSession();
    }
}

File: greendao_multitable/src/main/java/greendao/develop/android/multi/table/DBManager.kt
package greendao.develop.android.multi.table


import greendao.develop.android.multi.table.bean.one.DaoMaster
import greendao.develop.android.multi.table.bean.one.DaoSession

object DBManager {

    private const val SQL_NAME = "multi_table_sample"

    val daoSession: DaoSession =
        DaoMaster(
            DaoMaster.DevOpenHelper(
                MultiTableApp.context,
                SQL_NAME,
                null
            ).writableDatabase
        ).newSession()
}
